### DISCUSSION TITLE
Web Services SOAP over http

### DISCUSSION THREAD

**Discussion by testlorenzospa** [nextgenhealthcare/connect#5315] [Q&A] [answered]
Hello.

I am not very expert about HL7.

I must create in Mirth Connect a channel that can receive HL7 messages over http from a Web Services SOAP.

What kind of channel must I create?
A simple HTTP listener?
Or instead a Web Service listener?

Thanks.

**Comment by pacmano1** [comment, marked as answer]
If the entity sending you messages expects to make SOAP calls to you, then that is a a web service listener.  To secure that traffic, it should be using TLS.   Note that the free version of Mirth does not support TLS on listeners.   You will need to either license the core extension pack or provide you own TLS offloading solution with Stunnel, haproxy or similar software.

**Reply by sahil-kaushik** [reply]
Hi, 

Can you please help with a sample project to create a custom web service listener ?

**Reply by pacmano1** [reply]
@sahil-kaushik , please don't hijack threads

**Comment by testlorenzospa** [comment]
Hello @pacmano1 

Thanks for your explanation.
